src/Sign.Core/DataFormatSigners/AzureSignToolSigner.cs

```diff
@@ -113,25 +116,23 @@ public async Task SignAsync(IEnumerable<FileInfo> files, SignOptions options)
                 options.FileHashAlgorithm,
                 timestampConfiguration))
             {
+                FileInfo[] staThreadFiles = [.. files.Where(file => StaThreadExtensions.Contains(file.Extension))];
+                foreach (FileInfo file in staThreadFiles)
+                {
+                    await RunOnStaThread(() => SignAsync(signer, file, options));
```

Comment:
This only works because the actual signing operation executes synchronously before any `await` calls.  `SignAsync(...)` does contain an `await Thread.Delay(...)`, but it only executes after the first attempt.  

So, with the current implementation, only the first try has any chance of success while subsequent retries will fail because the executing thread will be on the thread pool (not STA).

I think the proper fix involves setting a synchronization context on the STA thread so that all continuations resume on the STA thread.

https://devblogs.microsoft.com/dotnet/await-synchronizationcontext-and-console-apps/
https://devblogs.microsoft.com/dotnet/await-synchronizationcontext-and-console-apps-part-2/